cumulusci/tasks/salesforce/install_package_version.py

```diff
@@ -82,18 +85,44 @@ def _init_options(self, kwargs):
                 "retry_interval_add"
             ]
 
-        # TODO: This should be centralized somewhere in the `dependencies` module,
-        # along with the same code working with `sources`.
-        # We're not using resolution strategies here - we could be,
-        # and this task could be a thin layer on top of update_dependencies.
-        if version == "latest":
-            self.options["version"] = str(self.project_config.get_latest_version())
-        elif version == "latest_beta":
-            self.options["version"] = str(
-                self.project_config.get_latest_version(beta=True)
+        dependency = None
+        github = f"https://github.com/{self.project_config.repo_owner}/{self.project_config.repo_name}"
+
+        if version in ["latest", "latest_beta"]:
+            strategy = "include_beta" if version == "latest_beta" else "production"
+            dependency = GitHubDynamicDependency(github=github)
+            dependency.resolve(
+                self.project_config, get_resolver_stack(self.project_config, strategy)
             )
         elif version == "previous":
-            self.options["version"] = str(self.project_config.get_previous_version())
+            release = find_previous_release(
+                self.project_config.get_repo(),
+                self.project_config.project__git__prefix_release,
+            )
+            dependency = GitHubDynamicDependency(github=github, tag=release.tag_name)
+            dependency.resolve(
+                self.project_config,
+                get_resolver_stack(self.project_config, "production"),
+            )
+
+        if dependency:
+            if dependency.managed_dependency:
+                # Handle 2GP and 1GP releases in a backwards-compatible way.
+                if isinstance(
+                    dependency.managed_dependency, PackageNamespaceVersionDependency
+                ):
+                    self.options["version"] = dependency.managed_dependency.version
+                elif isinstance(
+                    dependency.managed_dependency, PackageVersionIdDependency
+                ):
+                    self.options["version"] = dependency.managed_dependency.version_id
+                    self.options[
+                        "version_number"
+                    ] = dependency.managed_dependency.version_number
```

Comment:
I was a bit confused by why you're doing this rather than simply storing the dependency instance, but then I realized it needs to work like this in order to support freezing.